lib/jobs/OperationStatusPollerJob.js

```diff
@@ -41,63 +54,71 @@ class OperationStatusPollerJob extends BaseJob {
+
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return this
+            .getBrokerClient()
+            .getDeploymentBackupStatus(deploymentName, operationResp.token, boshDirectorName);
+        } else if (operationName === 'restore') {
+          return this
+            .getBrokerClient()
+            .getDeploymentRestoreStatus(deploymentName, operationResp.token, boshDirectorName);
+        } else {
+          throw new errors.BadRequest(`Operation ${operationName} not supported by status poller.`);
+        }
+      })
+      .then(operationStatusResponse => {
+        if (isFinished(operationStatusResponse.state)) {
+          return ScheduleManager.cancelSchedule(`${deploymentName}_${operationName}_${backupGuid}`, CONST.JOB.OPERATION_STATUS_POLLER)
+            .then(() => {
+              logger.info(`Deployment ${deploymentName} ${operationName} for backup guid ${backupGuid} completed -`, operationStatusResponse);
+              return operationStatusResponse;
+            });
+        } else {
+          logger.info(`Deployment ${deploymentName} ${operationName} for backup guid ${backupGuid} still in-progress - `, operationStatusResponse);
+          const currTime = moment();
+          // 'backup_restore_status_poller_timeout' config data might need to put in job data: operation specific
+          // operation can be other than backup/restore : thought just for future reference
+          if (currTime.diff(operationStartedAt) > config.backup.backup_restore_status_poller_timeout) {
+            return ScheduleManager.cancelSchedule(`${deploymentName}_${operationName}_${backupGuid}`, CONST.JOB.OPERATION_STATUS_POLLER)
               .then(() => {
-                logger.info(`Deployment ${deploymentName} ${operationName} for backup guid ${backupGuid} completed -`, operationStatusResponse);
-                if (operationStatusResponse.state === 'succeeded') {
-                  this.runSucceeded(operationStatusResponse, job, done);
-                } else {
-                  const msg = `Deployment ${deploymentName} ${operationName} for backup guid ${backupGuid} ${operationStatusResponse.state}`;
-                  logger.error(msg);
-                  const err = {
-                    statusCode: `ERR_DEPLOYMENT_${operationName.toUpperCase()}_${operationStatusResponse.state.toUpperCase()}`,
-                    statusMessage: msg
-                  };
-                  this.runFailed(err, operationStatusResponse, job, done);
-                }
+                const msg = `Deployment ${deploymentName} ${operationName} with backup guid ${backupGuid} exceeding timeout time 
+                      ${config.backup.backup_restore_status_poller_timeout / 1000 / 60} (mins). Stopping status check`;
+                logger.error(msg);
+                return operationStatusResponse;
```

Comment:
operationResponse must indicate the same status that the operation is being timed out. This is not being set in the same. Also, when the poller is timing out, it must abort the backup that might still be running in background and might finish eventually. Look at FabrikStatusPoller timeout [implementation](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/lib/fabrik/FabrikStatusPoller.js#L81).

Reply:
you must invoke abort method on the agents on the deployment & also, update backup status to aborted in meta data. If you can update the sate of metadata to aborted irrespective of the response from the agents on the box, then you can hit the same API even when deployment is not found rather than patching the status as aborted seperately. So same abort implementation can be invoked from here as well as NotFound flow.

Reply:
Are you proposing to bring in abort also in OOBBackupManager ? This was not in plan.